eng/scripts/vcpkg.ps1

```diff
@@ -0,0 +1,67 @@
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $Ref = 'master',
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Dependencies,
+
+    [Parameter()]
+    [string] $TargetPath ="$env:TEMP/$([guid]::NewGuid())",
+
+    [Parameter()]
+    [switch] $CI = ($null -ne $env:SYSTEM_TEAMPROJECTID)
+)
+
+function OutputWarning {
+    param([string] $Output)
+
+    if ($CI) {
+        Write-Host "##vso[task.logissue type=warning]$Output"
+    } else {
+        Write-Warning $Output
+    }
+}
+
+function SetEnvironmentVariable {
+    param(
+        [string] $Name,
+        [string] $Value
+    )
+
+    if ($CI) {
+        Write-Host "##vso[task.setvariable variable=$Name;]$($Value)"
+    } else {
+        Write-Verbose "Setting local environment variable: $Name = ***"
+        Set-Item -Path "env:$Name" -Value $Value
+    }
+}
+
+
+New-Item -ItemType Directory -Path $TargetPath -Force
+pushd $TargetPath
+
+git clone https://github.com/Microsoft/vcpkg .
+git checkout (Get-Content "$PSScriptRoot/../vcpkg-ref.txt")
```

Comment:
Shouldn't this be the `$Ref` you are passing in?

Reply:
We don't pass in a `ref` parameter anymore. This centralizes the location of the ref to `vcpkg-ref.txt` so it appears once in our codebase and isn't subject to shifting line placement.